website/docs/cli/commands/plan.mdx

```diff
@@ -290,6 +290,8 @@ The available options are:
   * 1 = Error
   * 2 = Succeeded with non-empty diff (changes present)
 
+- `-generate-config-out=PATH` - (Experimental) If import blocks are present in configuration, instructs Terraform to generate HCL for any imported resources not already present. The configuration is written to a new file at PATH, which must not already exist, or Terraform will error. If the plan fails for another reason, Terraform may still attempt to write configuration.
```

Comment:
general question -- will apply also support this and work seamlessly? we have a note on the apply page that it supports all plan options. i wonder if we need to give any addtl guidance on that

Reply:
This flag is only supported for the `plan` command. There are two categories of `plan` command flags: Plan Customization Options, which are shared with `apply`, and Other Options, which are not. Another example of an option supported by `plan` but not `apply` is `-detailed-exitcode`.

When generating configuration, we want users to run an explicit `plan` step and review the configuration before `apply`. So for example, it is not valid (or possible) to generate configuration during a `terraform apply -auto-approve` - if you try to do this with unmatched import blocks then it will fail and ask you to add config.